lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -92,6 +92,63 @@ class TensorEmptyConversionPattern
   }
 };
 
+class OnesOpConversionPattern : public OpConversionPattern<ttir::OnesOp> {
+public:
+  using OpConversionPattern<ttir::OnesOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::OnesOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // Get ttnn::TTNNLayoutAttr of the result type
+    //
+    ttnn::TTNNLayoutAttr layoutAttr = mlir::cast<ttnn::TTNNLayoutAttr>(
+        op.getResult().getType().getEncoding());
+
+    // Get the shape of tensor
+    //
+    // TODO(svuckovic): (#1435) ShapeAttr accepts int64_t, when it should be
+    // uint32_t
+    //
+    ttnn::ShapeAttr shapeAttr = ttnn::ShapeAttr::get(
+        rewriter.getContext(), llvm::SmallVector<int64_t, 4>(
+                                   op.getShape().begin(), op.getShape().end()));
+
+    // Get memref
+    //
+    mlir::MemRefType memref = layoutAttr.getMemref();
+
+    // Get data type, tensor layout, device and memory config
+    //
+    DataTypeAttr dTypeAttr = DataTypeAttr::get(
+        rewriter.getContext(), ttnn::utils::getDataTypeFromMemRef(memref));
+    ttnn::BufferType bufferType = layoutAttr.getBufferType();
+    ttnn::Layout ttnnLayoutEnum = llvm::isa<TileType>(memref.getElementType())
+                                      ? ttnn::Layout::Tile
+                                      : ttnn::Layout::RowMajor;
+    ttnn::LayoutAttr tensorLayoutAttr =
+        ttnn::LayoutAttr::get(op.getContext(), ttnnLayoutEnum);
+    ttnn::TensorMemoryLayoutAttr memLayout = layoutAttr.getMemLayout();
+
+    // Device only exists if ttnn::TensorMemoryLayout is None
+    //
+    auto device =
+        memLayout ? nullptr : ::ttnn::utils::getOrInsertDevice(rewriter, op);
```

Comment:
if `memLayout` is enum, then check `memLayout != None`.

Reply:
Check https://github.com/tenstorrent/tt-mlir/commit/357853836b0a00ba296ceb8d2c679bc8886638d7